app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -45,185 +44,211 @@ import com.wire.android.ui.common.dimensions
+            text = text,
+            onDismiss = { featureFlagNotificationViewModel.hideDialogStatus() },
+            optionButton1Properties = WireDialogButtonProperties(
+                onClick = { featureFlagNotificationViewModel.hideDialogStatus() },
+                text = stringResource(id = R.string.label_ok),
+                type = WireDialogButtonType.Primary,
             )
-        }
+        )
+    }
+
+    HomeContent(
+        connectivityState = commonTopAppBarViewModel.connectivityState,
+        homeState = homeViewModel.homeState,
+        homeScreenState = homeScreenState,
+        conversationListState = conversationListViewModel.conversationListState,
+        onReturnToCallClick = commonTopAppBarViewModel::openOngoingCallScreen,
+        onNewConversationClick = conversationListViewModel::openNewConversation,
+        onSelfUserClick = homeViewModel::navigateToSelfUserProfile,
+        navigateToItem = homeViewModel::navigateTo
+    )
+
+    BackHandler(homeScreenState.searchBarState.isSearchActive) {
+        homeScreenState.searchBarState.closeSearch()
     }
 }
 
 @OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class, ExperimentalAnimationApi::class)
 @Composable
 fun HomeContent(
-    homeBottomSheetState: ModalBottomSheetState,
-    snackbarHostState: SnackbarHostState,
-    homeBottomSheetContent: @Composable (ColumnScope.() -> Unit)?,
-    currentNavigationItem: HomeNavigationItem,
-    homeNavigationGraph: @Composable () -> Unit,
-    homeTopBar: @Composable (elevation: Dp) -> Unit,
-    openNewConversation: () -> Unit,
+    connectivityState: ConnectivityUIState,
+    homeState: HomeState,
+    homeScreenState: HomeScreenState,
     conversationListState: ConversationListState,
-    navController: NavHostController,
+    onReturnToCallClick: () -> Unit,
+    onNewConversationClick: () -> Unit,
+    onSelfUserClick: () -> Unit,
+    navigateToItem: (NavigationItem) -> Unit
 ) {
-    with(currentNavigationItem) {
-        WireModalSheetLayout(
-            sheetState = homeBottomSheetState,
-            coroutineScope = rememberCoroutineScope(),
-            // we want to render "nothing" instead of doing a if/else check
-            // on homeBottomSheetContent and wrap homeContent() into WireModalSheetLayout
-            // or render it without WireModalSheetLayout to avoid
-            // recomposing the homeContent() when homeBottomSheetContent
-            // changes from null to "something"
-            sheetContent = homeBottomSheetContent ?: { }
-        ) {
-            CollapsingTopBarScaffold(
-                snapOnFling = false,
-                keepElevationWhenCollapsed = true,
-                topBarHeader = homeTopBar,
-                snackbarHost = {
-                    SwipeDismissSnackbarHost(
-                        hostState = snackbarHostState,
-                        modifier = Modifier.fillMaxWidth()
+    with(homeScreenState) {
+        with(homeState) {
```

Comment:
Both objects are mixed because both are accessed via `this` and you don't know which method and field come from which one, for instance if you use `hashCode()` then you need to remember that you're inside the inner `with` and if you want to differentiate them by using `this@with.hashCode()` then you get "There is more than one label with such a name in this scope", so yeah, it works but it's confusing and can be error-prone. 😉

Reply:
I agree, I used only the first with